lib/streamlit/elements/doc_string.py

```diff
@@ -86,66 +125,416 @@ def _marshall(doc_string_proto: DocStringProto, obj: Any) -> None:
+        # secrets.toml file.
+        return None
+
+    if inspect.isclass(value) or inspect.ismodule(value) or callable(value):
+        return None
+
+    value_str = repr(value)
+
+    if isinstance(value, str):
+        # Special-case strings as human-readable because they're allowed to look like
+        # "<foo blarg at 0x15ee6f9a0>".
+        return _shorten(value_str)
+
+    if is_mem_address_str(value_str):
+        # If value_str looks like "<foo blarg at 0x15ee6f9a0>" it's not human readable.
+        return None
+
+    return _shorten(value_str)
+
+
+def _shorten(s, length=300):
+    s = s.strip()
+    return s[:length] + "..." if len(s) > length else s
+
+
+def _is_computed_property(obj, attr_name):
+    obj_class = getattr(obj, "__class__", None)
+
+    if not obj_class:
+        return False
+
+    # Go through superclasses in order of inheritance (mro) to see if any of them have an
+    # attribute called attr_name. If so, check if it's a @property.
+    for parent_class in inspect.getmro(obj_class):
+        class_attr = getattr(parent_class, attr_name, None)
+
+        if class_attr is None:
+            continue
+
+        # If is property, return it.
+        if isinstance(class_attr, property) or inspect.isgetsetdescriptor(class_attr):
+            return True
+
+    return False
+
+
+def _get_members(obj):
+    members_for_sorting = []
+
+    for attr_name in dir(obj):
+        if attr_name.startswith("_"):
+            continue
+
+        is_computed_value = _is_computed_property(obj, attr_name)
+
+        if is_computed_value:
+            parent_attr = getattr(obj.__class__, attr_name)
+
+            member_type = "property"
+
+            weight = 0
+            member_docs = _get_docstring(parent_attr)
+            member_value = None
+        else:
+            attr_value = getattr(obj, attr_name)
+            weight = _get_weight(attr_value)
+
+            human_readable_value = _get_human_readable_value(attr_value)
+
+            member_type = _get_type_as_str(attr_value)
+
+            if human_readable_value is None:
+                member_docs = _get_docstring(attr_value)
+                member_value = None
+            else:
+                member_docs = None
+                member_value = human_readable_value
+
+        if member_type == "module":
+            # Don't pollute the output with all imported modules.
+            continue
+
+        member = MemberProto()
+        member.name = attr_name
+        member.type = member_type
+
+        if member_docs is not None:
+            member.doc_string = _get_first_line(member_docs)
+
+        if member_value is not None:
+            member.value = member_value
+
+        members_for_sorting.append((weight, member))
+
+    if members_for_sorting:
+        _, members = zip(*sorted(members_for_sorting, key=lambda x: (x[0], x[1].name)))
+        return members
+
+    return []
```

Comment:
nit: I think this can be rewritten to something a bit more readable to those unfamiliar with the `zip` function and splat operator by writing something like

```python3
sorted_members = sorted(members_for_sorting, key=lambda x: (x[0], x[1].name))
return [m for _, m in sorted_members]
```